### PR TITLE
Remove StarknetEthProxyCheck

### LINKED ISSUE #1277: Try to change ProxyChecks in order to works with more proxy contracts

### PR DESCRIPTION
**Pull request by tkumor3**
Closes #1277 

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/guide/resolving_proxy_contracts.rst [hunk 1]
```diff
@@ -37,6 +37,10 @@ By default, ``proxy_config`` uses a configuration with two **ProxyChecks**:
  - ArgentProxyCheck - resolves `Argent Proxy <https://github.com/argentlabs/argent-contracts-starknet/blob/b7c4af7462a461386d29551400b985832ba942de/contracts/upgrade/Proxy.cairo>`_.
  - OpenZeppelinProxyCheck - resolves `OpenZeppelin Proxy <https://github.com/OpenZeppelin/cairo-contracts/blob/d12abf335f5c778fd19d6f99e91c099b40865deb/src/openzeppelin/upgrades/presets/Proxy.cairo>`_.
 
+.. warning::
+
+    ``StarknetEthProxyCheck`` has been removed, because StarkGate ETH Token contract changes the type of interaction. Currently, StarkGate * Token contracts should be used as ERC20.
```
Comment (ddoktorski) [inline docs/guide/resolving_proxy_contracts.rst:42]: What do you mean exactly by `changes the type of interaction`? I believe we should simplify this information, emphasizing that the ETH contract has been updated (to Cairo 2) and as it's no longer a Proxy, there's no need to use `proxy_config` in the `Contract` class.

**Review by ddoktorski — COMMENTED** [review]
Please also mention this change in the migration guide.

File: docs/guide/resolving_proxy_contracts.rst (hunk 1 above)
Comment (tkumor3) [reply docs/guide/resolving_proxy_contracts.rst:42]: `type of interaction` was taken from Starkscan, but they changed the description of `Type` row! Your proposition is better

File: docs/migration_guide.rst [hunk 2]
```diff
@@ -1,6 +1,16 @@
 Migration guide
 ===============
 
+**********************
+0.20.0 Migration guide
+**********************
+
+0.20.0 Deprecations
+-------------------
+.. currentmodule::  ~starknet_py.proxy.proxy_check
```
Comment (ddoktorski) [inline docs/migration_guide.rst:10]: nit: It can be removed

File: docs/migration_guide.rst [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+
+- :class:`StarknetEthProxyCheck` has been removed, because StarkGate ETH contract isn't proxy anymore
```
Comment (ddoktorski) [inline docs/migration_guide.rst:12]:
```suggestion
- :class:`StarknetEthProxyCheck` has been removed from the Proxy checks
```

File: docs/migration_guide.rst [hunk 4]
```diff
@@ -1,6 +1,16 @@
 Migration guide
 ===============
 
+**********************
+0.20.0 Migration guide
+**********************
+
+0.20.0 Deprecations
```
Comment (ddoktorski) [inline docs/migration_guide.rst:8]: I would consider categorizing this under `Breaking Changes`. Technically, it's mostly a breaking change on the Starknet side, yet it definitely doesn't classify as a deprecation.

File: docs/migration_guide.rst [hunk 5]
```diff
@@ -1,6 +1,15 @@
...
 ===============
 
+**********************
+0.20.0 Migration guide
+**********************
+
+0.20.0 Breaking Changes
+-----------------------
+
+- :class:`StarknetEthProxyCheck` has been removed from the Proxy checks
```
Comment (ddoktorski) [inline docs/migration_guide.rst:11]: nit: Can be changed in other PR
```suggestion
1. :class:`StarknetEthProxyCheck` has been removed from the Proxy checks
```

**Review by ddoktorski — APPROVED** [review]
🔥

**Pull request merged**